### ISSUE TITLE
Creation of domain results in provider referencing existing subdomain

### ISSUE THREAD

**Issue by sempervictus** [ClouDNS/terraform-provider-cloudns#6]
A prior project in a separate environment created `subdomain.ourdomain.net` and records within it - everything works as expected.
Today we spun up a project which needs to create records in `ourdomain.net` and defined it the same way as a `cloudns_dns_zone`. The `cloudns_dns_zone` is created but the statefile ends up referencing the `cloudns_dns_zone` **from the prior project** and all `cloudns_dns_record` attempts then fail with:
```hcl
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for cloudns_dns_record.ourdomain-net-a["somerecord"] to include new values learned so far during apply, provider "registry.opentofu.org/cloudns/cloudns" produced an invalid new value for .zone: was cty.StringVal("ourdomain.net"),
│ but now cty.StringVal("subdomain.ourdomain.net").
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```
we also cannot fix the state because
```hcl
╷
│ Error: resource cloudns_dns_zone doesn't support import
│ 
│ 
╵
```

**Comment by Venkoul99** [comment]
Hello @sempervictus , is it possible to provide both projects parameters so we can test further and fix the issues?

**Comment by Venkoul99** [comment]
Hello, the issue with the subdomains now should be fixed. We have released a new version v1.0.5 of the terraform provider.

**Comment by Venkoul99** [comment]
We'll go ahead and close this issue for now. If you encounter any further problems or have any other questions, please don't hesitate to reach out to us again.
